main.go

```diff
@@ -16,5 +16,123 @@
 
 package main
 
+import (
+	"context"
+	"net"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	nested "github.com/antonfisher/nested-logrus-formatter"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/endpoint"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/kernel"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/ipam/point2pointipam"
+	"github.com/networkservicemesh/sdk/pkg/tools/debug"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/signalctx"
+	"github.com/networkservicemesh/sdk/pkg/tools/spiffejwt"
+)
+
+// config
+type Config struct {
```

Comment:
Should we also consider the registration in the registry? I guess we can do here something like this: https://github.com/networkservicemesh/integration-k8s-kind/blob/master/cmd/nse/main.go#L113

Reply:
Yeah... a similar idea occurred to me actually... because it would let this be much more generically usable for various purposes.